syncer/dml.go

```diff
@@ -419,14 +419,13 @@ func genKeyList(table string, columns []*model.ColumnInfo, dataSeq []interface{}
 			log.L().Debug("ignore null value", zap.String("column", columns[i].Name.O), zap.String("table", table))
 			continue // ignore `null` value.
 		}
-		// for key, I think no need to add the `,` separator.
-		buf.WriteString(columnValue(data, &columns[i].FieldType))
+		// one column key looks like:`column_val.column_name.`
+		buf.WriteString(columnValue(data, &columns[i].FieldType) + "." + columns[i].Name.String() + ".")
```

Comment:
Since we have used buf.WriteString,  no need to use '+'?

Reply:
fixed in https://github.com/pingcap/dm/pull/2043/commits/15591572207e44183a75659551635368341e2ad6